nf_core/modules/install.py

```diff
@@ -5,137 +5,174 @@
         super().__init__(pipeline_dir)
         self.force = force
         self.latest = latest
         self.sha = sha
+        self.update_all = update_all
 
     def install(self, module):
         if self.repo_type == "modules":
             log.error("You cannot install a module in a clone of nf-core/modules")
             return False
         # Check whether pipelines is valid
         self.has_valid_directory()
+        if not self.update_all:
+            # Get the available modules
+            try:
+                self.modules_repo.get_modules_file_tree()
+            except LookupError as e:
+                log.error(e)
+                return False
 
-        # Get the available modules
-        try:
-            self.modules_repo.get_modules_file_tree()
-        except LookupError as e:
-            log.error(e)
-            return False
-
-        if self.latest and self.sha is not None:
-            log.error("Cannot use '--sha' and '--latest' at the same time!")
-            return False
-
-        if module is None:
-            module = questionary.autocomplete(
-                "Tool name:",
-                choices=self.modules_repo.modules_avail_module_names,
-                style=nf_core.utils.nfcore_question_style,
-            ).unsafe_ask()
+            if self.latest and self.sha is not None:
+                log.error("Cannot use '--sha' and '--latest' at the same time!")
+                return False
 
-        # Check that the supplied name is an available module
-        if module not in self.modules_repo.modules_avail_module_names:
-            log.error("Module '{}' not found in list of available modules.".format(module))
-            log.info("Use the command 'nf-core modules list' to view available software")
-            return False
+            if module is None:
+                module = questionary.autocomplete(
+                    "Tool name:",
+                    choices=self.modules_repo.modules_avail_module_names,
+                    style=nf_core.utils.nfcore_question_style,
+                ).unsafe_ask()
 
-        # Set the install folder based on the repository name
-        install_folder = [self.modules_repo.owner, self.modules_repo.repo]
+            # Check that the supplied name is an available module
+            if module and module not in self.modules_repo.modules_avail_module_names:
+                log.error("Module '{}' not found in list of available modules.".format(module))
+                log.info("Use the command 'nf-core modules list' to view available software")
+                return False
+            repos_and_modules = [(self.modules_repo, module)]
+        else:
+            if module:
+                raise UserWarning("You cannot specify a module and use '--all' flag at the same time")
+            if self.force:
+                log.warning("'--force' doesn't do anything when used with '--all'")
+            else:
+                self.force = True
```

Comment:
```suggestion
            self.force = True
```
We could also log that info but I feel like it's not necessary and makes the code a bit simpler when not adding that if-statement. But you decide :)